### ISSUE TITLE
Detailed implementation for splitting train and test data

### ISSUE THREAD

**Issue by sh0416** [sh0416/bpr#3]
In ML dataset, All data are visited at most one time.

However, in the other dataset, a item might be referenced multiple times by one user.

In this case, how to handle when Item i is both inside the training set and the test set?

Do I need to remove the item i from the test set?

If I remove the item i, then evaluation metric represents the predictive power of the model which has not seen yet. However, it does not capture the repetitive pattern inside a user.

If The item i is not removed from the test set, then evaluation metric could become vulnerable to overfitting. However, it could represent the repetitive pattern.

The current implementation excludes the observed item, i.e. item from the training set, during splitting data and evaluation process.

To make an additional option for this consideration will be a good solution for handling this issue.
